pkg/generate/app/templatelookup_test.go

```diff
@@ -0,0 +1,113 @@
+package app
+
+import (
+	"testing"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	ktestclient "k8s.io/kubernetes/pkg/client/unversioned/testclient"
+	"k8s.io/kubernetes/pkg/runtime"
+
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/client/testclient"
+	templateapi "github.com/openshift/origin/pkg/template/api"
+)
+
+func testTemplateClient(templates *templateapi.TemplateList) client.Interface {
+	fake := &testclient.Fake{}
+	fake.AddReactor("list", "templates", func(action ktestclient.Action) (handled bool, ret runtime.Object, err error) {
+		if len(action.GetNamespace()) > 0 {
+			matchingTemplates := &templateapi.TemplateList{
+				Items: []templateapi.Template{},
+			}
+			for _, template := range templates.Items {
+				if template.Namespace == action.GetNamespace() {
+					matchingTemplates.Items = append(matchingTemplates.Items, template)
+				}
+			}
+			return true, matchingTemplates, nil
+		} else {
+			return true, templates, nil
+		}
+	})
+	return fake
+}
+
+func TestTemplateSearcher(t *testing.T) {
+	testData := map[string][]string{
+		"openshift":   []string{"nodejs-mongodb-example", "rails-postgresql-example", "jenkins-ephemeral", "my-jenkins"},
+		"mynamespace": []string{"my-jenkins"},
+	}
+
+	tests := []struct {
+		value           string
+		expectedMatches int
+		expectedErr     bool
+	}{
+		{
+			value:           "jenkins",
+			expectedMatches: 3,
+		},
+		{
+			value:           "my-jenkins",
+			expectedMatches: 2,
+		},
+		{
+			value:           "jenkins-ephemeral",
+			expectedMatches: 1,
+		},
```

Comment:
how about a test that looks for "my-jenkins/jenkins-ephemeral" to confirm it gets no result when the template only exists in the openshift namespace and not the referenced namespace?